dolphinscheduler-task-plugin/dolphinscheduler-task-java/src/main/java/org/apache/dolphinscheduler/plugin/task/java/JavaTask.java

```diff
@@ -0,0 +1,422 @@
+     **/
+    protected void createJavaSourceFileIfNotExists(String sourceCode, String fileName) throws IOException {
+        logger.info("tenantCode: {}, task dir:{}", taskRequest.getTenantCode(), taskRequest.getExecutePath());
+
+        if (!Files.exists(Paths.get(fileName))) {
+            logger.info("generate java source file: {}", fileName);
+
+            StringBuilder sb = new StringBuilder();
+            sb.append(sourceCode);
+            logger.info(sb.toString());
+
+            // write data to file
+            FileUtils.writeStringToFile(new File(fileName),
+                    sb.toString(),
+                    StandardCharsets.UTF_8);
+        } else {
+            throw new JavaSourceFileExistException("java source file exists, please report an issue on official.");
+        }
+    }
+
+    /**
+     * @description: Construct the full path name of the Java source file from the temporary execution path of the task
+     * @param: []
+     * @return: java.lang.String
+     **/
+    protected String buildJavaSourceCodeFileFullName(String publicClassName) {
+        return String.format(JavaConstants.JAVA_SOURCE_CODE_NAME_TEMPLATE, taskRequest.getExecutePath(), publicClassName);
+    }
+
+    /**
+     * @description: Construct a Classpath or module path based on isModulePath
+     * @param: []
+     * @return: java.lang.String
+     **/
+    protected String buildResourcePath() {
+        StringBuilder builder = new StringBuilder();
+        if (javaParameters.isModulePath()) {
+            builder.append("--module-path");
+        } else {
+            builder.append("--class-path");
+        }
+        builder.append(" ").append(JavaConstants.CLASSPATH_CURRENT_DIR)
+                .append(JavaConstants.PATH_SEPARATOR)
+                .append(taskRequest.getExecutePath());
+        for (ResourceInfo info : javaParameters.getResourceFilesList()) {
+            builder.append(JavaConstants.PATH_SEPARATOR);
+            builder.append(taskRequest.getExecutePath())
+                    .append(info.getResourceName());
+        }
+        return builder.toString();
+    }
+
+    /**
+     * @description: Compile the Java source file
+     * @param: [java.lang.String]
+     * @return: java.lang.String
+     **/
+    protected String compilerRawScript(String sourceCode) throws IOException, InterruptedException {
+        String publicClassName = getPublicClassName(sourceCode);
+        String fileName =  buildJavaSourceCodeFileFullName(publicClassName);
+        createJavaSourceFileIfNotExists(sourceCode, fileName);
+        String compileCommand = buildJavaCompileCommand(fileName, sourceCode);
+        Preconditions.checkNotNull(compileCommand, "command not be null.");
+        TaskResponse compileResponse = shellCommandExecutor.run(compileCommand);
+        // must drop the command file ,if do not ,the next command will not run. because be limited the ShellCommandExecutor's create file rules
+        dropShellCommandFile();
```

Comment:
This should be a common logic for shell task?

Reply:
thx, It is a bit strange to have two processes running, I will optimize to run in one process